Source/WebKit/UIProcess/ios/forms/WKDateTimeInputControl.mm

```diff
@@ -67,31 +77,47 @@ @implementation WKDateTimePicker
 static NSString * const kMonthFormatString = @"yyyy-MM"; // "2011-01".
 static NSString * const kTimeFormatString = @"HH:mm"; // "13:45".
 static NSString * const kDateTimeFormatString = @"yyyy-MM-dd'T'HH:mm"; // "2011-01-27T13:45"
+static NSString * const kWeekFormatString = @"yyyy-'W'ww";
 static constexpr auto yearAndMonthDatePickerMode = static_cast<UIDatePickerMode>(4269);
 
-- (id)initWithView:(WKContentView *)view datePickerMode:(UIDatePickerMode)mode
+- (id)initWithView:(WKContentView *)view inputType:(WebKit::InputType)inputType
 {
     if (!(self = [super init]))
         return nil;
 
     _view = view;
     _interactionPoint = [_view lastInteractionLocation];
+    UIDatePickerMode mode;
 
-    switch (view.focusedElementInformation.elementType) {
+    switch (inputType) {
     case WebKit::InputType::Date:
+        mode = UIDatePickerModeDate;
         _formatString = kDateFormatString;
         break;
     case WebKit::InputType::Month:
+        mode = yearAndMonthDatePickerMode;
         _formatString = kMonthFormatString;
         break;
     case WebKit::InputType::Time:
+        mode = UIDatePickerModeTime;
         _formatString = kTimeFormatString;
         break;
     case WebKit::InputType::DateTimeLocal:
+        mode = UIDatePickerModeDateAndTime;
         _formatString = kDateTimeFormatString;
         break;
+#if HAVE(UICALENDARSELECTIONWEEKOFYEAR)
+    case WebKit::InputType::Week:
+        _formatString = kWeekFormatString;
+        _selectionWeekOfYear = adoptNS([[UICalendarSelectionWeekOfYear alloc] initWithDelegate:self]);
+        _calendarView = adoptNS([[UICalendarView alloc] init]);
+        [_calendarView setCalendar:[NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601]];
+        [_calendarView setSelectionBehavior:_selectionWeekOfYear.get()];
+        return self;
+#endif
     default:
-        break;
+        [self release];
+        return nil;
```

Comment:
Is this change necessary?

Reply:
It was changed because otherwise `mode` would be used uninitialized since at that point it didn't match any of the targeted input types. Would it be better to set it some value anyways?

Reply:
I see! I think it's fine then, we don't expect this code to run anyway.